### PR TITLE
Disables Thread-Safe for local static variables

### LINKED ISSUE #1283: Exit status 3221225477 returned from node scripts/build.js / Windows XP
# Windows XP after update node.js to 5 version trying to install grunt-sass <h2>
<a href="https://gist.github.com/FrontEndMonster/085d5d35e9d21f31f681">npm debug log</a>

"devDependencies": {
    "autoprefixer-core": "^5.2.1",
    "grunt": "^0.4.5",
    "grunt-browser-sync": "2.1.3",
    "grunt-cli": "0.1.13",
    "grunt-combine-media-queries": "1.0.19",
    "grunt-contrib-clean": "0.6.0",
    "grunt-contrib-copy": "^0.8.0",
    "grunt-contrib-imagemin": "1.0.0",
    "grunt-contrib-watch": "0.6.1",
    "grunt-csscomb": "3.0.1",
    "grunt-newer": "1.1.1",
    "grunt-notify": "0.4.1",
    "grunt-pngmin": "1.0.2",
    "grunt-postcss": "^0.5.5",
    "grunt-sass": "1.1.0",
    "grunt-sftp-deploy": "0.2.4",
    "grunt-svg2string": "0.1.6",
    "grunt-svgmin": "^2.0.1",
    "grunt-svgstore": "0.5.0",
    "load-grunt-tasks": "3.2.0"
  }

17930 warn EPACKAGEJSON grunt-with-node-sass@0.9.8 No description
17931 warn EPACKAGEJSON grunt-with-node-sass@0.9.8 No repository field.
17932 warn EPACKAGEJSON grunt-with-node-sass@0.9.8 No license field.
17933 verbose stack Error: node-sass@3.4.2 postinstall: `node scripts/build.js`
17933 verbose stack Exit status 3221225477
17933 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:233:16)
17933 verbose stack     at emitTwo (events.js:87:13)
17933 verbose stack     at EventEmitter.emit (events.js:172:7)
17933 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
17933 verbose stack     at emitTwo (events.js:87:13)
17933 verbose stack     at ChildProcess.emit (events.js:172:7)
17933 verbose stack     at maybeClose (internal/child_process.js:818:16)
17933 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
17934 verbose pkgid node-sass@3.4.2
17935 verbose cwd E:\ProxySocks\markup
17936 error Windows_NT 5.1.2600
17937 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "node-sass"
17938 error node v5.0.0
17939 error npm  v3.3.6
17940 error code ELIFECYCLE
17941 error node-sass@3.4.2 postinstall: `node scripts/build.js`
17941 error Exit status 3221225477
17942 error Failed at the node-sass@3.4.2 postinstall script 'node scripts/build.js'.
17942 error This is most likely a problem with the node-sass package,
17942 error not with npm itself.
17942 error Tell the author that this fails on your system:
17942 error     node scripts/build.js
17942 error You can get their info via:
17942 error     npm owner ls node-sass
17942 error There is likely additional logging output above.
17943 verbose exit [ 1, true ]

### PR DESCRIPTION
**Pull request by am11**
Fixes #1283.

### CODE REVIEW

**Comment by am11** [comment]
@xzyfer, @saper, PTAL.

**Comment by saper** [comment]
I don't think this is a good idea for one reason. As long as the node releases are built with VS2013 we have to keep compatibility with C++. I don't want to waste 3 days on yet-another-C++-runtime incompatibility the next time we encounter something.

We should simply build with the same compiler and possibly with the same flags as the node engine.

I'll be submitting patch to node and node-gyp to let us know the original PlatformToolset used, so we can possibly use VS2015 in `v120` mode to compile node-sass.

**Comment by am11** [comment]
The cflag is only applied to 2015. This doesn't effect VS2013.

**Comment by saper** [comment]
I understand. We should really use PlatformToolset to set things straight on VS2015 (if that is confirmed working, I didn't try yet).

**Comment by am11** [comment]
@saper, I don't think you are understanding it correctly
v120 = VS2013
v140 = VS2015.
If someone has BuiltTools2015 installed, they will not be able to build node-sass because it doesn't have v120 at all. If someone has VS2015 installed without VS2013 toolset (which is optional in the install menu), they will missed out too. Having a compiler flag in gyp is not yet-another-hassle.

**Comment by saper** [comment]
On Wed, 20 Jan 2016, Adeel Mujahid wrote:

> @saper, I don't think you are understanding it correctly
> v120 = VS2013
> v140 = VS2015.
> If someone has BuiltTools2015 installed, they will not be able to build node-sass because it doesn't have v120 at all. If someone has VS2015 installed without VS2013 toolset (which is optional in the install menu), they will missed out too. Having a compiler flag in gyp is not yet-another-hassle.

We actually need to tell people using node-gyp to use vs2013 or vs2015
with platformtoolset v120 (which is as you say
the C++ compiler taken out of vs2013).

I cannot have C++ compatibility otherwise and that's what we are using.

**Comment by am11** [comment]
> We actually need to tell people using node-gyp to use vs2013 or vs2015

If someone has VS2015 installed without legacy stuff or BuildTools2015, they will not have v120 targets.

**Comment by saper** [comment]
On Wed, 20 Jan 2016, Adeel Mujahid wrote:

> > We actually need to tell people using node-gyp to use vs2013 or vs2015
> 
> If someone has VS2015 installed without legacy stuff or BuildTools2015, they will not have v120 targets.

In other words: I care less about people installing VS2015 with the default or limited settings
and compiling node-sass on their own

I care a lot more about folk using our binaries as is.

**Comment by am11** [comment]
This change should not effect our binaries built from VS2013.

**Comment by saper** [comment]
On Wed, 20 Jan 2016, Adeel Mujahid wrote:

> This change should not effect our binaries built from VS2013.

Let me be a bit blunt here:

I've wasted enough time to work on the "assumed" VS2015 compatibility.
Things were supposed to run smooth, they didn't.

I don't really care if somebody at the Microsoft tried node-sass
personally with VS2015 or not. They haven't obviously used XP
for their testing.

If Microsoft comes with a statement "We do support dynamic loading
of C++ DLLs compiled with Visual Studio 2015 into the address space
of the process runing C++ code compiled with Visual Studio 2013"
then I am fine.

The problem is, nobody at Microsoft will give us such a guarantee
for C++. They may guarantee compatibility for pure C modules only.

Keep in mind that

https://connect.microsoft.com/VisualStudio/feedback/details/1789709/visual-c-2015-runtime-broken-on-windows-server-2003-c-11-magic-statics

got closed as "(works) by design". The workaround is given for this particular
C++ feature.

As for this patch, I don't know why we are setting globally
that npm variable. Building node modules with v140 is simply
wrong, as long as v120 is what node is using.

I am not going to elaborate further on this, sorry.

**Comment by am11** [comment]
TLDR: the change in appveyor was not meant for this PR. I have reverted it back. #1352 will take care of AppVeyor. This change remain with zero side effects.
